services/QuillLMS/app/workers/batch_save_activity_session_concept_results_worker.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class BatchSaveActivitySessionConceptResultsWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: SidekiqQueue::CRITICAL
+
+  def perform(concept_results, activity_session_id, activity_session_uid)
+    batch_runner(activity_session_uid) do
+      concept_results
+        .map { |cr| concept_results_hash(cr, activity_session_id) }
+        .reject(&:empty?)
+        .each { |crh| SaveActivitySessionConceptResultsWorker.perform_async(crh) }
+    end
+  end
+
+  def on_success(_status, options)
```

Comment:
This is more of a product question, but the batch has a `complete` and a `success` callback. If one of the 15 jobs fails, do you want to show nothing (`success`) or show the results of the 14 (`complete`). We might want `complete` here, but it's more of a product question than an engineering one.

Reply:
@dandrabik that's a good callout. I suspect we want complete rather than success (which would be closer to the existing behavior, anyway, since currently we don't rely at all on the concept results actually *being* saved). I'll update this.